src/gleam/list.gleam

```diff
@@ -706,8 +733,17 @@ pub fn flatten(lists: List(List(a))) -> List(a) {
 /// ```
 ///
 pub fn flat_map(over list: List(a), with fun: fn(a) -> List(b)) -> List(b) {
-  map(list, fun)
-  |> flatten
+  flat_map_loop(list, fun, [])
+}
+
+fn flat_map_loop(list: List(a), fun: fn(a) -> List(b), acc: List(b)) -> List(b) {
+  case list {
+    [] -> reverse(acc)
+    [first, ..rest] -> {
+      let list = fun(first)
+      flat_map_loop(rest, fun, reverse_and_prepend(list, to: acc))
```

Comment:
would this read better?
```
first |> fun |> reverse_and_prepend(to: acc) |> flat_map_loop(rest, fun, _)
```

Reply:
What happens for 100k or 1mio for `new.max`?

Reply:
Nah I think this is fine already